ArjunaJTA/recovery/src/main/java/org/jboss/narayana/jta/quickstarts/recovery/JmsRecovery.java

```diff
@@ -270,86 +240,35 @@ private int consumeMessages(MessageConsumer consumer, long millis, int cnt) thro
     }
 
     public int consumeMessages(int cnt, long millis) throws Exception {
-        XAConnection connection = xacf.createXAConnection();
+        
         int msgCnt = 0;
 
-        try
-        {
+        try(XAConnection connection = xacf.createXAConnection();
+                XASession xaSession = connection.createXASession();
+                MessageConsumer xaConsumer = xaSession.createConsumer(queue);) {
```

Comment:
Yep, I see your point here. My only concern is whether this is a functional or a style change. I'd lean toward functional here but that's up to you